### PR TITLE
Rebalance Dragon Knight 2024-02-05

### PR DESCRIPTION
**Pull request by Atemo**
* **Addressed Issue(s)**: https://github.com/rathena/rathena/issues/8130

* **Server Mode**: renewal

* **Description of Pull Request**: 

Dragon Knight
-------------------------

1. Servant Weapon
	- Reduces skill cooldown from 150 seconds to 60 seconds based on level 5.
	- Increases base damage from 2500%Atk to 4850%Atk based on level 5.
	- Increases number of hit from 2 hits to 3 hits.
	- No longer be removed by Dispell or Clearance.
	- Increases the trigger chance of weapon from 15% to 25% based on level 5.

2. Madness Crusher
	- Increases base damage from 8350%Atk to 20000%Atk based on level 5.

3. Storm Slash
	- Increases base damage from 2200%Atk to 4050%Atk per hit based on level 5.

4. Hack and Slasher
	- Increases base damage from 7700%Atk to 8550%Atk per hit based on level 10.

5. Dragonic Breath
	- Increases factor weight of MaxHP in skill formula from 12.5% to 25%.
	- Increases factor weight of MaxSP in skill formula from 25% to 50%.

### CODE REVIEW

**Comment by datawulf** [comment]
In case I can help, that should be the Dragonic Breath Formula
```
		case DK_DRAGONIC_BREATH:
			skillratio += -100 + 50 + 350 * skill_lv + 7 * sstatus->pow;
			if (sc && sc->getSCE(SC_DRAGONIC_AURA))
				skillratio += (sstatus->max_hp * 175 / 10000 + status_get_max_sp(src) * 35 / 1000) * skill_lv;
			else 
				skillratio += (sstatus->max_hp * 125 / 10000 + status_get_max_sp(src) * 25 / 1000) * skill_lv;
			RE_LVL_DMOD(100);
			break;
```

**Review by aleos89 — APPROVED** [review]

File: src/map/battle.cpp [hunk 1]
```diff
@@ -9719,7 +9725,7 @@ enum damage_lv battle_weapon_attack(struct block_list* src, struct block_list* t
 		}
 
 		if( sc ){
-			if( sc->getSCE( SC_SERVANTWEAPON ) && sd->servantball > 0 && rnd() % 100 < ( 3 * sc->getSCE( SC_SERVANTWEAPON )->val1 ) ){
+			if( sc->getSCE( SC_SERVANTWEAPON ) && sd->servantball > 0 && rnd() % 100 < ( 5 * sc->getSCE( SC_SERVANTWEAPON )->val1 ) ){
```
Comment (Lemongrass3110) [inline src/map/battle.cpp:9728]:
```suggestion
			if( sc->getSCE( SC_SERVANTWEAPON ) && sd->servantball > 0 && rnd_chance( 5 * sc->getSCE( SC_SERVANTWEAPON )->val1, 100 ) ){
```

File: src/map/battle.cpp [hunk 2]
```diff
@@ -5456,7 +5456,7 @@ static int battle_calc_attack_skill_ratio(struct Damage* wd, struct block_list *
 			RE_LVL_DMOD(100);
 			break;
 		case DK_MADNESS_CRUSHER:
-			skillratio += -100 + 350 + 1600 * skill_lv + 10 * sstatus->pow;
+			skillratio += -100 + 1000 + 3800 * skill_lv + 10 * sstatus->pow;
```
Comment (Lemongrass3110) [inline src/map/battle.cpp:5459]:
```suggestion
			skillratio += -100 + 1000 + 3800 * skill_lv;
			skillratio += 10 * sstatus->pow;
```

File: src/map/battle.cpp [hunk 3]
```diff
@@ -5469,18 +5469,24 @@ static int battle_calc_attack_skill_ratio(struct Damage* wd, struct block_list *
 				skillratio *= 2;
 			break;
 		case DK_STORMSLASH:
-			skillratio += -100 + 200 + 400 * skill_lv + 5 * sstatus->pow;
+			skillratio += -100 + 300 + 750 * skill_lv + 5 * sstatus->pow;
```
Comment (Lemongrass3110) [inline src/map/battle.cpp:5472]:
```suggestion
			skillratio += -100 + 300 + 750 * skill_lv;
			skillratio += 5 * sstatus->pow;
```

File: src/map/battle.cpp [hunk 4]
```diff
@@ -5432,7 +5432,7 @@ static int battle_calc_attack_skill_ratio(struct Damage* wd, struct block_list *
 				skillratio += skillratio * sc->getSCE(SC_LIGHTOFSTAR)->val2 / 100;
 			break;
 		case DK_SERVANTWEAPON_ATK:
-			skillratio += -100 + 500 + 400 * skill_lv + 5 * sstatus->pow;
+			skillratio += -100 + 450 + 800 * skill_lv + 5 * sstatus->pow;
```
Comment (Lemongrass3110) [inline src/map/battle.cpp:5435]:
```suggestion
			skillratio += -100 + 450 + 800 * skill_lv;
			skillratio += 5 * sstatus->pow;
```

**Review by Lemongrass3110 — APPROVED** [review]

**Pull request merged**